tests/boardfarm_plugins/boardfarm_prplmesh/devices/prplmesh_compose.py

```diff
@@ -0,0 +1,115 @@
+# SPDX-License-Identifier: BSD-2-Clause-Patent
+# SPDX-FileCopyrightText: 2020 the prplMesh contributors (see AUTHORS.md)
+# This code is subject to the terms of the BSD+Patent license.
+# See LICENSE file for more details.
+
+import os
+import time
+
+import boardfarm
+from environment import ALEntityDocker, _get_bridge_interface
+from .prplmesh_base import PrplMeshBase
+from sniffer import Sniffer
+
+rootdir = os.path.abspath(os.path.join(os.path.dirname(__file__), '../../../..'))
+
+
+class PrplMeshCompose(PrplMeshBase):
+    """Dockerized prplMesh device."""
+
+    model = ("prplmesh_compose")
+    agent_entity = None
+    controller_entity = None
+
+    def __init__(self, *args, **kwargs):
+        self.args = args
+        self.kwargs = kwargs
+
+        config = kwargs.get("config", kwargs)
+
+        # List of device's consoles test can interact with
+        self.consoles = [self]
+
+        # Getting unic ID to distinguish devices and network they belong to
+        self.unique_id = os.getenv("RUN_ID")
+        self.user_id = os.getenv("SUDO_USER", os.getenv("USER", ""))
+
+        self.docker_name = "-".join((config.get("name", "prplmesh_compose"), self.unique_id))
+        self.name = config.get("name", "prplmesh_compose")
```

Comment:
_Nitpick_: reorder of lines will allow just to use `self.name` without additional parse of `config`. Improves readability.
```suggestion
        self.name = config.get("name", "prplmesh_compose")
        self.docker_name = "-".join((self.name, self.unique_id))
```

Also: usage of `prplmesh_compose` as `self.name` may break the tests. Currently, in the test we are locating devices by `self.name` ([example](https://github.com/prplfoundation/prplMesh/blob/master/tests/boardfarm_plugins/boardfarm_prplmesh/tests/ap_config_renew.py#L18)). To referrer devices by name, it should be among members of boardfarm build-in [enum](https://github.com/mattsm/boardfarm/blob/master/boardfarm/lib/DeviceManager.py#L25).
But i would suggest just to abort the test is `name` isn't set in configuration file.

Reply:
Fixed on 356793b1